src/libraries/System.Text.Json/tests/Serialization/ExtensionDataTests.cs

```diff
@@ -1066,5 +1066,53 @@ public override void Write(Utf8JsonWriter writer, JsonElement value, JsonSeriali
                 writer.WriteEndObject();
             }
         }
+
+        [Theory]
+        [InlineData(typeof(HasExtensionData<Dictionary<string, JsonElement>>))]
+        [InlineData(typeof(HasExtensionData<SortedDictionary<string, JsonElement>>))]
+        [InlineData(typeof(HasExtensionData<IDictionary<string, JsonElement>>))]
+        [InlineData(typeof(HasExtensionData<IReadOnlyDictionary<string, JsonElement>>))]
+        [InlineData(typeof(HasExtensionData<ImmutableDictionary<string, JsonElement>>))]
+        [InlineData(typeof(HasExtensionData<ImmutableSortedDictionary<string, JsonElement>>))]
+        [InlineData(typeof(HasExtensionData<IImmutableDictionary<string, JsonElement>>))]
+        [InlineData(typeof(HasExtensionData<ConcurrentDictionary<string, JsonElement>>))]
+        public static void TypeValidityAsExtensionDataWithJsonElementValue(Type type)
+        {
+            const string json = "{\"foo\": 123}";
+
+            object obj = JsonSerializer.Deserialize(json, type);
+            IDictionary<string, JsonElement> extData = (IDictionary<string, JsonElement>)type.GetProperty("ExtensionData").GetValue(obj)!;
+            Assert.True(extData.ContainsKey("foo"));
+
+            int value = extData["foo"].GetInt32();
+            Assert.Equal(123, value);
+        }
+
+        [Theory]
+        [InlineData(typeof(HasExtensionData<Dictionary<string, object>>))]
```

Comment:
We need negative tests where the generic dictionary types are supported, but they key type is not `string` and/or the value type is not `object` or `JsonElement`.

Reply:
The negative tests we have as `ExtensionProperty_InvalidDictionary` but we can add tests for the specific types if you think that's worthwhile.

Do you mean non generic dictionaries like `IDictionary`? I don't think that will work since it needs to match either `<string, object>` or `<string, JsonElement>`.